orangecontrib/prototypes/widgets/owexplpred.py

```diff
@@ -0,0 +1,401 @@
+
+            perm = np.random.choice([True, False], batchMxSize, replace=True)
+            perm = np.reshape(perm, (self.batchSize, noAtr))
+            rand_data = self.data.X[np.random.randint(
+                dataRows, size=self.batchSize), :]
+            inst1.X = np.copy(tiled_inst.X)
+            inst1.X[perm] = rand_data[perm]
+            inst2.X = np.copy(inst1.X)
+
+            inst1.X[:, a] = tiled_inst.X[:, a]
+            inst2.X[:, a] = rand_data[:, a]
+            f1 = self._get_predictions(inst1, classValue)
+            f2 = self._get_predictions(inst2, classValue)
+
+            diff = np.sum(f1 - f2)
+            expl[0, a] += diff
+
+            # update variance
+            steps[0, a] += self.batchSize
+            iterations_reached[0, a] += self.batchSize
+            d = diff - mu[0, a]
+            mu[0, a] += d / steps[0, a]
+            M2[0, a] += d * (diff - mu[0, a])
+            var[0, a] = M2[0, a] / (steps[0, a] - 1)
+
+            if (callback()):
+                break
+
+            # exclude from sampling if necessary
+            neededIter = zSq * var[0, a] / (self.error**2)
+            if (neededIter <= steps[0, a]) and (steps[0, a] >= self.minIter) or (steps[0, a] > self.maxIter):
+                iterations_reached[0, a] = self.maxIter + 1
+                atr_err[0, a] = np.sqrt(zSq * var[0, a] / steps[0, a])
+
+        expl[0, :] = expl[0, :]/steps[0, :]
+
+        # creating return array
+        domain = Domain([self.atr_names], [
+                        ContinuousVariable('contributions'), ContinuousVariable('errors')])
+        table = Table.from_list(domain, np.asarray(
+            self.atr_names.values).reshape(-1, 1))
+        table.Y[:, 0] = expl.T[:, 0]
+        table.Y[:, 1] = atr_err.T[:, 0]
+        table.X = table.X
+        return classValue, table
+
+    def _get_predictions(self, inst, classValue):
+        if isinstance(self.data.domain.class_vars[0], ContinuousVariable):
+            # regression
+            return self.model(inst)
+        else:
+            # classification
+            predictions = (self.model(inst) == classValue) * 1
+            return predictions
+
+
+class OWExplainPred(OWWidget):
+
+    name = "Explain Predictions"
+    description = "Computes attribute contributions to the final prediction with an approximation algorithm for shapely value"
+    #icon = "iconImage.png"
+    priority = 200
+    gui_error = settings.Setting(5)
+    gui_p_val = settings.Setting(5)
+
+    class Inputs:
+        data = Input("Data", Table, default=True)
+        model = Input("Model", Model, multiple=False)
+        sample = Input("Sample", Table)
+
+    class Outputs:
+        explanations = Output("Explanations", Table)
+
+    class Warning(OWWidget.Warning):
+        empty_data = widget.Msg("Empty dataset.")
+        sample_too_big = widget.Msg("Too many samples to explain.")
+        selection_not_matching = widget.Msg(
```

Comment:
These two should be Errors not Warnings since the widget does nothing in those cases.